extensions-api/src/main/scala/pl/touk/nussknacker/engine/deployment/CustomActionDefinition.scala

```diff
@@ -26,7 +28,13 @@ case class CustomActionDefinition(
     icon: Option[URI] = None
 )
 
-//TODO: validators, defaultValue, hint, labelOpt?
-case class CustomActionParameter(name: String, editor: ParameterEditor)
+case class CustomActionParameter(name: String, editor: ParameterEditor, validators: Option[List[ParameterValidator]])
 
 case class CustomActionResult(msg: String)
+
+sealed trait CustomActionValidationResult
+
+object CustomActionValidationResult {
```

Comment:
You should use `Validated` and `ValidatedNel` where Valid and Invalid are already implemented.